src/api-types.ts

```diff
@@ -1064,3 +1129,36 @@ export type UpdatePropertySchema =
   | UpdateMultiSelectPropertySchema
   | RenamePropertySchema
   | null
+
+/*
+ * Files
+ */
+
+interface FileBase {
+  type: "file"
+  file: {
+    url: string
+  }
+}
+
+export type File = FileBase & { file: { expiry_time: string } }
+
+export type FileInput = Omit<FileBase, "type"> & { type?: "file" }
+
+export interface ExternalFile {
+  type: "external"
+  external: {
+    url: string
+  }
+}
+
+export type ExternalFileInput = Omit<ExternalFile, "type"> & {
```

Comment:
not blocking but I wonder if it is helpful to add a type helper to make `type` optional (stack overflow [example](https://stackoverflow.com/questions/43159887/make-a-single-property-optional-in-typescript))

Reply:
Turns out we have `PartialBy` which does the same thing 🥳